R/validate.R

```diff
@@ -62,6 +62,10 @@ validate_input_values <- function(lgd_senior_claims, lgd_subordinated_claims,
   # ADO 1943 - Once we decide to add a separate Merton calculation on the average
   # maturity of a portfolio, this check will need to be removed
   if (!term %% 1 == 0) {
-    stop("Argmuent term must be a whole number")
+    stop("Argmemnt term must be a whole number")
+  }
+
+  if (!asset_type %in% asset_types_lookup) {
+    stop("Invalid value for argument asset_type")
   }
```

Comment:
> Use stop(call. = FALSE), rlang::abort(), Rf_errorcall(R_NilValue, ...) to avoid cluttering the error message with the name of the function that generated it. That information is often not informative, and can easily be accessed via traceback() or an IDE equivalent.

-- https://style.tidyverse.org/error-messages.html

Reply:
will do in context of logging/signalling overview ticket shortyl